core/finality-grandpa/src/environment.rs

```diff
@@ -731,7 +750,11 @@ where
 		equivocation: Equivocation<Self::Id, Precommit<Block>, Self::Signature>
 	) {
 		warn!(target: "afg", "Detected precommit equivocation in the finality worker: {:?}", equivocation);
-		// nothing yet
+		submit_report_call(
+			&self.inner,
+			&self.transaction_pool,
+			Call::Grandpa(GrandpaCall::report_precommit_equivocation(equivocation.encode())),
```

Comment:
prevote equivocations take the `(set_id, round)` as well. That seems right to me -- any reason it's only `equivocation` here?